### PR TITLE
Add validation rule UniqueEnumValueNames

### PR DESCRIPTION
**Pull request by spawnia**
https://github.com/graphql/graphql-js/commit/0737212567a394caccc92f26a0705869f9e4ee3b

### CODE REVIEW

**Comment by spawnia** [comment]
@vhenzl working on this, just need to finish tests

File: src/Validator/Rules/UniqueEnumValueNames.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+{
+    public function getSDLVisitor(SDLValidationContext $context): array
+    {
+        /** @var array<string, array<string, EnumValueNode>> $knownValueNames */
+        $knownValueNames = [];
+
+        /**
+         * @param (Node&EnumTypeDefinitionNode)|(Node&EnumTypeExtensionNode) $enum
+         */
+        $checkValueUniqueness = static function ($enum) use ($context, &$knownValueNames): VisitorOperation {
```
Comment (vhenzl) [inline src/Validator/Rules/UniqueEnumValueNames.php:28]:
```suggestion
        $checkValueUniqueness = static function (Node $enum) use ($context, &$knownValueNames): VisitorOperation {
```
Native type here?

File: tests/Validator/UniqueEnumValueNamesTest.php [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+            new UniqueEnumValueNames(),
+            '
+      extend enum SomeEnum {
+        FOO
+      }
+      extend enum SomeEnum {
+        FOO
+      }
+        ',
+            BuildSchema::build('
```
Comment (vhenzl) [inline tests/Validator/UniqueEnumValueNamesTest.php:260]: I would prefer to build the schema before calling expect and pass `$schema` here. That way the test structure would be more similar to `graphql-js` and the code easier to compare. 

Applies also to the tests above.

**Review by vhenzl — COMMENTED** [review]
Two suggestions. Looks good to me in general.

File: src/Validator/Rules/UniqueEnumValueNames.php (hunk 1 above)
Comment (vhenzl) [reply src/Validator/Rules/UniqueEnumValueNames.php:28]: See https://github.com/webonyx/graphql-php/pull/995#discussion_r741482567

Comment (spawnia) [reply src/Validator/Rules/UniqueEnumValueNames.php:27]: I would feel safer with a native type, but am fine with relying on tests proving the correctness of this piece of code. I would rather not worsen performance.

**Comment by simPod** [comment]
💯 0

**Pull request merged**